kratos/containers/pointer_vector_set.h

```diff
@@ -1066,51 +1066,52 @@ class PointerVectorSet final
             return;
         }
 
-        if (empty()) {
-            mData.reserve(std::distance(first, last));
+        if (empty() || (KeyOf(end() - 1) < KeyOf(GetReference(first)))) {
+            mData.reserve(mData.size() + std::distance(first, last));
             for (auto it = first; it != last; ++it) {
                 mData.push_back(TPointerType(&GetReference(it)));
```

Comment:
Seems that we created a memory issue in the latest changes! We cannot create a new shared pointer out of the reference. This will cause a early release of the memory